### PR TITLE
Refactoring c make lists for unit tests

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
These changes include next tests into a build: 
- config_profile_test
- connection_handler_test
- formatters_test
- hmi_message_handler_test
- media_manager_test
- protocol_handler_test
- resumption_test
- rpc_base_test
- security_manager_test
- smart_object_test
- telemetry_monitor_test
- transport_manager_test

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft @AByzhynar @LuxoftAKutsan @SNiukalov Please review this PR.

File: src/components/config_profile/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     DESTINATION
+     ${CMAKE_CURRENT_BINARY_DIR})
+file(COPY
+     smartDeviceLink_invalid_int.ini
+     DESTINATION
+     ${CMAKE_CURRENT_BINARY_DIR})
+file(COPY
+     smartDeviceLink_invalid_boolean.ini
+     DESTINATION
+     ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (AByzhynar) [inline src/components/config_profile/test/CMakeLists.txt:45]: @BSolonenko Why did you divide copying of all files to separate functions?

**Review by LuxoftAKutsan — APPROVED** [review]

File: src/components/transport_manager/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+target_link_libraries(${PROJECT_NAME}
+                      PRIVATE transport_manager::transport_manager
+                              components::interface)
+
+if(BUILD_BT_SUPPORT)
+  target_link_libraries(${PROJECT_NAME} PRIVATE bluetooth)
+endif()
+
+if(BUILD_USB_SUPPORT)
```
Comment (mkorniichuk) [inline src/components/transport_manager/test/CMakeLists.txt:60]: @BSolonenko maybe use single if?

File: src/components/transport_manager/test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+set(PLATFORM_DEPENDENT_SOURCES)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  target_sources(${PROJECT_NAME} PRIVATE platform_specific/linux)
+endif()
+list(APPEND SOURCES ${PLATFORM_DEPENDENT_SOURCES})
+
+target_sources(${PROJECT_NAME}
+               PRIVATE
+               ${CMAKE_CURRENT_LIST_DIR}/network_interface_listener_test.cc
```
Comment (mkorniichuk) [inline src/components/transport_manager/test/CMakeLists.txt:42]: @BSolonenko Why not to use **collect_sources**?

File: src/components/telemetry_monitor/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+project(telemetry_monitor_test)
+create_test(${PROJECT_NAME})
+
+target_include_directories(${PROJECT_NAME}
+                           PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+target_sources(${PROJECT_NAME}
+               PRIVATE
+               ${CMAKE_CURRENT_LIST_DIR}/application_manager_metric_test.cc
```
Comment (mkorniichuk) [inline src/components/telemetry_monitor/test/CMakeLists.txt:36]: @BSolonenko I believe, **collect_sources** would help

File: src/components/security_manager/test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+set(CERT_LIST ${CMAKE_SOURCE_DIR}/mycert.pem ${CMAKE_SOURCE_DIR}/mykey.pem)
+
+add_custom_target(
+  generate_certificates
+  COMMAND ${CMAKE_SOURCE_DIR}/tools/Utils/generate_test_certificates.py
+          -d=${CMAKE_CURRENT_BINARY_DIR} -s
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+
+foreach(CERT_ENTRY ${CERT_LIST})
+  file(COPY ${CERT_ENTRY} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (mkorniichuk) [inline src/components/security_manager/test/CMakeLists.txt:58]: @BSolonenko have you tried to pass a list as an argument?

File: src/components/config_profile/test/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+file(COPY smartDeviceLink.ini DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (mkorniichuk) [inline src/components/config_profile/test/CMakeLists.txt:28]: @BSolonenko how about *.ini?

**Review by mkorniichuk — CHANGES_REQUESTED** [review]

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/config_profile/test/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2018, Ford Motor Company All rights reserved.
```
Comment (AByzhynar) [inline src/components/config_profile/test/CMakeLists.txt:1]: @BSolonenko 2019, isn't it?

File: src/components/connection_handler/test/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2018, Ford Motor Company All rights reserved.
```
Comment (AByzhynar) [inline src/components/connection_handler/test/CMakeLists.txt:1]: @BSolonenko 2019, isn't it?

File: src/components/formatters/test/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2018, Ford Motor Company All rights reserved.
```
Comment (AByzhynar) [inline src/components/formatters/test/CMakeLists.txt:1]: @BSolonenko 2019, isn't it?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**